### ISSUE TITLE
Incorrect `popXX` macro assigned for X86 `POP Rmr16` & `POP Rmr32` instructions in `ia.sinc`.

### ISSUE THREAD

**Issue by Wall-AF** [NationalSecurityAgency/ghidra#7326]
The 2 lines below were incorrectly changed in https://github.com/NationalSecurityAgency/ghidra/commit/6d7dc046f023f0c7b2f2956f62d21f086ba216d5 to
https://github.com/NationalSecurityAgency/ghidra/blob/14527b015e09172bc4a24423fe3bc11391ef6af9/Ghidra/Processors/x86/data/languages/ia.sinc#L3737-L3738
instead of
```
:POP Rmr16      is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & row=5 & page=1 & Rmr16       { local val:2 = 0; pop42(val); Rmr16 = val; }
:POP Rmr32      is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=1 & row=5 & page=1 & Rmr32       { local val:4 = 0; pop24(val); Rmr32 = val; }
```
causing incorrect decompilation.

**Comment by Wall-AF** [comment]
While you're at it check the definitions for:
https://github.com/NationalSecurityAgency/ghidra/blob/14527b015e09172bc4a24423fe3bc11391ef6af9/Ghidra/Processors/x86/data/languages/ia.sinc#L3816-L3817
as there should be 4 versions:
```
:PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push22(tmp); }
:PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push42(tmp); }
:PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push24(tmp); }
:PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push44(tmp); }
```

**Comment by Wall-AF** [comment]
Also missing from the `<output>` section of the `__cdecl16far` prototype
https://github.com/NationalSecurityAgency/ghidra/blob/a1f243ebbab2fbad75cdd6e76b463902523b4ead/Ghidra/Processors/x86/data/languages/x86-16.cspec#L114-L125
is
```
      <pentry minsize="3" maxsize="4">
        <addr space="join" piece1="DX" piece2="AX"/>
      </pentry>
```

**Comment by jcranmer** [comment]
> While you're at it check the definitions for:
> 
> [ghidra/Ghidra/Processors/x86/data/languages/ia.sinc](https://github.com/NationalSecurityAgency/ghidra/blob/14527b015e09172bc4a24423fe3bc11391ef6af9/Ghidra/Processors/x86/data/languages/ia.sinc#L3816-L3817)
> 
> Lines 3816 to 3817 in [14527b0](/NationalSecurityAgency/ghidra/commit/14527b015e09172bc4a24423fe3bc11391ef6af9)
>  :PUSH simm16_16    is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=0 & byte=0x68; simm16_16    { tmp:2=simm16_16; push22(tmp); } 
>  :PUSH simm16_16    is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & byte=0x68; simm16_16    { tmp:2=simm16_16; push42(tmp); } 
> 
> as there should be 4 versions:
> 
> ```
> :PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push22(tmp); }
> :PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push42(tmp); }
> :PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push24(tmp); }
> :PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push44(tmp); }
> ```

I was just about to report this as a separate bug, it's nice to see that someone already reported it.

(Strictly speaking, it's not based on the instruction address size but the stack address size in the `ss` register. But there's no functionality in the sleigh semantics for stack address size parameter, and it's probably a reasonable assumption that addrsize = stackaddrsize for most practical code).

**Comment by Wall-AF** [comment]
> While you're at it check the definitions for:
> 
> [ghidra/Ghidra/Processors/x86/data/languages/ia.sinc](https://github.com/NationalSecurityAgency/ghidra/blob/14527b015e09172bc4a24423fe3bc11391ef6af9/Ghidra/Processors/x86/data/languages/ia.sinc#L3816-L3817)
> 
> Lines 3816 to 3817 in [14527b0](/NationalSecurityAgency/ghidra/commit/14527b015e09172bc4a24423fe3bc11391ef6af9)
>  :PUSH simm16_16    is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=0 & byte=0x68; simm16_16    { tmp:2=simm16_16; push22(tmp); } 
>  :PUSH simm16_16    is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & byte=0x68; simm16_16    { tmp:2=simm16_16; push42(tmp); } 
> 
> as there should be 4 versions:
> 
> ```
> :PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push22(tmp); }
> :PUSH simm8_16     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=0 & byte=0x6a; simm8_16     { tmp:2=simm8_16; push42(tmp); }
> :PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=0 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push24(tmp); }
> :PUSH simm8_32     is $(LONGMODE_OFF) & vexMode=0 & addrsize=1 & opsize=1 & byte=0x6a; simm8_32     { tmp:4=simm8_32; push44(tmp); }
> ```

@ryanmkurtz sorry, but you've missed these!

**Comment by ryanmkurtz** [comment]
Deferring to @GhidorahRex

**Comment by GhidorahRex** [comment]
@Wall-AF Yes, can you make those a separate issue? I'll get a fix in for them
